source/require-load-differences.txt

```diff
@@ -0,0 +1,131 @@
+Differences Between ``require()`` and ``load()``
+================================================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+The ``require()`` and ``load()`` methods include files and modules in
+your scripts for added functionality. However, ``require()`` and
+``load()`` differ in their behaviors and availability.
+
+Types of Scripts in ``mongosh``
+-------------------------------
+
+You can use the following types of scripts with ``mongosh``:
+
+- ``mongosh`` scripts, which can be any of the following:
+
+  - Code entered directly into the REPL.
+  - The :ref:`mongoshrc.js <mongoshrc-js>` file.
+  - Code loaded with the :method:`load()` method.
+
+- Node.js scripts, which are any scripts loaded with ``require()``,
+  including npm packages. These scripts are always files.
+
+Availability of ``require()`` and ``load()``
+--------------------------------------------
+
+The ``require()`` and ``load()`` methods differ in availability
+depending on the type of script you are using.
+
+- In ``mongosh`` scripts, both ``require()`` and ``load()`` are available.
+
+- In Node.js scripts, only ``require()`` is available.
+
+File Paths for ``require()`` and ``load()``
+-------------------------------------------
+
+The type of script determines how you specify file paths with
+``require()`` or ``load()``.
+
+- In ``mongosh`` scripts:
+
+  - ``require()`` uses the standard
+    `Node.js module resolution algorithm
+    <https://nodejs.org/api/modules.html>`__, starting from the current
+    working directory of the shell.
+
+  - ``load()`` takes either:
+  
+    - An absolute path, or
+    - A relative path. When using a relative path, the path is always
+      interpreted as relative to the current working directory of the
+      shell.
+
+- In Node.js scripts, ``require()`` uses the standard
+  `Node.js module resolution algorithm
+  <https://nodejs.org/api/modules.html>`__, starting from the file
+  where ``require()`` was called.
+
+.. tip::
+
+   To return the current working directory of the shell, run the
+   :method:`pwd()` method from your script.
+
+   To change the shell's working directory, use the :method:`cd()`
+   method in your script.
+
+Load External Code in a ``mongosh`` Script
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can load external code in a ``mongosh`` script file, such as an npm
+package or a separate ``mongosh`` script.
+
+- To load a ``mongosh`` script from another ``mongosh`` script, use
+  the ``__dirname`` environment variable. The ``__dirname`` environment
+  variable returns the absolute path of the directory containing the
+  file being executed.
+
+  .. example::
+
+     To load a ``mongosh`` script named ``test-suite.js`` from another
+     ``mongosh`` script, add the following line to your script:
+
+     .. code-block:: javascript
+
+       load(__dirname + '/test-suite.js')
```

Comment:
@addaleax , @davemungo raised a question about this example that we wanted to run by you:

> Why would you prepend the absolute path to the current directory when load() can use relative paths? 

I tried to incorporate the example you provided in the JIRA ticket, so wasn't sure if I maybe misunderstood or if there's something we should adjust about the docs here.

Reply:
The problem is that `load()` inteprets path relative to the current working directory of the `mongosh` process.

That’s fine if you load scripts directly from the shell, because you (proably) know where that working directory is, but if you load a script from another script, you want to be sure that the script you’re loading does not depend on external factors like where `mongosh` was started.

Reply:
That makes sense, thank you for the explanation!